homeassistant/components/fibaro/__init__.py

```diff
@@ -619,49 +608,50 @@ def call_set_color(self, red, green, blue, white):
         blue = int(max(0, min(255, blue)))
         white = int(max(0, min(255, white)))
         color_str = f"{red},{green},{blue},{white}"
-        self.fibaro_device.properties.color = color_str
+        self.fibaro_device.properties["color"] = color_str
         self.action("setColor", str(red), str(green), str(blue), str(white))
 
     def action(self, cmd, *args):
         """Perform an action on the Fibaro HC."""
         if cmd in self.fibaro_device.actions:
-            getattr(self.fibaro_device, cmd)(*args)
+            self.fibaro_device.execute_action(cmd, args)
             _LOGGER.debug("-> %s.%s%s called", str(self.ha_id), str(cmd), str(args))
         else:
             self.dont_know_message(cmd)
 
     @property
     def current_binary_state(self):
         """Return the current binary state."""
-        if self.fibaro_device.properties.value == "false":
+        if self.fibaro_device.properties.get("value") == "false":
             return False
         if (
-            self.fibaro_device.properties.value == "true"
-            or int(self.fibaro_device.properties.value) > 0
+            self.fibaro_device.properties.get("value") == "true"
+            or int(self.fibaro_device.properties.get("value")) > 0
         ):
             return True
         return False
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes of the device."""
-        attr = {"fibaro_id": self.fibaro_device.id}
+        attr = {"fibaro_id": self.fibaro_device.fibaro_id}
 
-        try:
-            if "battery" in self.fibaro_device.interfaces:
-                attr[ATTR_BATTERY_LEVEL] = int(
-                    self.fibaro_device.properties.batteryLevel
-                )
-            if "armed" in self.fibaro_device.properties:
-                armed = self.fibaro_device.properties.armed
-                if isinstance(armed, bool):
-                    attr[ATTR_ARMED] = armed
-                elif isinstance(armed, str) and armed.lower() in ("true", "false"):
-                    attr[ATTR_ARMED] = armed.lower() == "true"
-                else:
-                    attr[ATTR_ARMED] = None
-        except (ValueError, KeyError):
-            pass
+        if isinstance(self.fibaro_device, DeviceModel):
+            try:
+                if "battery" in self.fibaro_device.raw_data.get("interfaces"):
+                    attr[ATTR_BATTERY_LEVEL] = int(
+                        self.fibaro_device.properties.get("batteryLevel")
+                    )
+                if "armed" in self.fibaro_device.properties:
+                    armed = self.fibaro_device.properties.get("armed")
+                    if isinstance(armed, bool):
+                        attr[ATTR_ARMED] = armed
+                    elif isinstance(armed, str) and armed.lower() in ("true", "false"):
+                        attr[ATTR_ARMED] = armed.lower() == "true"
+                    else:
+                        attr[ATTR_ARMED] = None
```

Comment:
This is a good example of what the library user shouldn't need to do. The library user should be able to access an attribute of the library to get the battery level and another attribute to get the armed state and those should be clearly typed without ambiguity.